core/src/main/java/org/wso2/msf4j/internal/websocket/EndpointDispatcher.java

```diff
@@ -0,0 +1,165 @@
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.msf4j.internal.websocket;
+
+import org.wso2.msf4j.websocket.exception.WebSocketEndpointAnnotationException;
+
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import javax.websocket.OnClose;
+import javax.websocket.OnError;
+import javax.websocket.OnMessage;
+import javax.websocket.OnOpen;
+import javax.websocket.PongMessage;
+import javax.websocket.server.ServerEndpoint;
+
+/**
+ * Dispatch the registered endpoints.
+ * This class will find the best matching resource of an endpoint for a given message type.
+ */
+public class EndpointDispatcher {
+
+    /**
+     * @return the URI of the Endpoint as a String.
+     * @throws WebSocketEndpointAnnotationException throws if {@link ServerEndpoint} is not found.
+     */
+    public String getUri(Object webSocketEndpoint) throws WebSocketEndpointAnnotationException {
+        ServerEndpoint serverEndpointAnnotation = webSocketEndpoint.getClass().getAnnotation(ServerEndpoint.class);
+        if (serverEndpointAnnotation == null) {
+            throw new WebSocketEndpointAnnotationException("ServerEndpoint is not declared");
+        }
+        return serverEndpointAnnotation.value();
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle new connection if exists else null.
+     */
+    public Method getOnOpenMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnOpen.class)) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle connection closure if exists else null.
+     */
+    public Method getOnCloseMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        if (methods == null) {
+            return null;
+        }
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnClose.class)) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
+     * @return method to handle errors if exists else null.
+     */
+    public Method getOnErrorMethod(Object webSocketEndpoint) {
+        Method[] methods = webSocketEndpoint.getClass().getMethods();
+        if (methods == null) {
+            return null;
+        }
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(OnError.class)) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param webSocketEndpoint Endpoint to extract method.
```

Comment:
Method description is missing. Explain what the method does in the first line ending with a full stop